compute/snapshots/delete_snapshots_by_filter.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_snapshot_delete_by_filter]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"google.golang.org/api/iterator"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// Deletes ALL disk snapshots in the project that match the filter
+func deleteByFilter(w io.Writer, projectID, filter string) error {
+	// projectID := "your_project_id"
+
+	// Learn more about filters:
+	// https://cloud.google.com/python/docs/reference/compute/latest/google.cloud.compute_v1.types.ListSnapshotsRequest
+	// filter := ""
```

Comment:
NIT: Since this variable is passed in to the function and there's another developer-provided variable (`projectID`), I would move this comment to be right after `projectID`.